core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -274,49 +325,225 @@ private void maybeResetInputAndSeekToEndIntermediateTopicOffsets(final Map consu
+                    System.out.println("Topic: " + topicPartition.topic());
                 }
             }
-            if (!dryRun) {
-                client.seekToEnd(intermediateTopicPartitions);
+
+            client.seekToEnd(intermediateTopicPartitions);
+        }
+    }
+
+    private void maybeReset(final String groupId,
+                            final Consumer<byte[], byte[]> client,
+                            final Set<TopicPartition> inputTopicPartitions)
+        throws Exception {
+
+        if (inputTopicPartitions.size() > 0) {
+            System.out.println("Following input topics offsets will be reset to (for consumer group " + groupId + ")");
+            if (options.has(toOffsetOption)) {
+                resetOffsetsTo(client, inputTopicPartitions, options.valueOf(toOffsetOption));
+            } else if (options.has(toEarliestOption)) {
+                client.seekToBeginning(inputTopicPartitions);
+            } else if (options.has(toLatestOption)) {
+                client.seekToEnd(inputTopicPartitions);
+            } else if (options.has(shiftByOption)) {
+                shiftOffsetsBy(client, inputTopicPartitions, options.valueOf(shiftByOption));
+            } else if (options.has(toDatetimeOption)) {
+                final String ts = options.valueOf(toDatetimeOption);
+                final Long timestamp = getDateTime(ts);
+                resetToDatetime(client, inputTopicPartitions, timestamp);
+            } else if (options.has(byDurationOption)) {
+                final String duration = options.valueOf(byDurationOption);
+                final Duration durationParsed = DatatypeFactory.newInstance().newDuration(duration);
+                resetByDuration(client, inputTopicPartitions, durationParsed);
+            } else if (options.has(fromFileOption)) {
+                final String resetPlanPath = options.valueOf(fromFileOption);
+                final Map<TopicPartition, Long> topicPartitionsAndOffset = getTopicPartitionOffsetFromResetPlan(resetPlanPath);
+                resetOffsetsFromResetPlan(client, inputTopicPartitions, topicPartitionsAndOffset);
+            } else {
+                client.seekToBeginning(inputTopicPartitions);
+            }
+
+            for (final TopicPartition p : inputTopicPartitions) {
+                final Long position = client.position(p);
+                System.out.println("Topic: " + p.topic() + " Partition: " + p.partition() + " Offset: " + position);
             }
         }
     }
 
-    private void maybeSeekToBeginning(final KafkaConsumer<byte[], byte[]> client,
-                                      final Set<TopicPartition> inputTopicPartitions) {
+    // visible for testing
+    public void resetOffsetsFromResetPlan(Consumer<byte[], byte[]> client, Set<TopicPartition> inputTopicPartitions, Map<TopicPartition, Long> topicPartitionsAndOffset) {
+        final Map<TopicPartition, Long> endOffsets = client.endOffsets(inputTopicPartitions);
```

Comment:
These two calls boils down to
```
Fetcher#retrieveOffsetsByTimes(Map<TopicPartition, Long> timestampsToSearch ...)
```
I wonder if they can be combined (to save roundtrips).

Reply:
We want to get `endOffsets()` and `beginningOffsets` for the same set of partitions. A single request cannot get both at once AFAIK.

Also, the reset tool is not considered to be on the "hot code path" -- thus, we don't need to worry about performance too much and apply (unnecessary?) micro optimizations.

Reply:
Agreed that such optimization is not needed for this task.